app/controllers/charge.controller.js

```diff
@@ -35,8 +35,29 @@ const { views, preserveProperties } = require('../../config/charge.controller')
 const { CORRELATION_HEADER } = require('../../config/correlation-header')
 const { createChargeIdSessionKey } = require('../utils/session')
 const { getLoggingFields } = require('../utils/logging-fields-helper')
+const applePayEnabled = (process.env.APPLE_PAY_ENABLED || 'true') === 'true'
+const googlePayEnabled = (process.env.GOOGLE_PAY_ENABLED || 'true') === 'true'
+const payTestGatewayAccounts = (process.env.PAY_TEST_GATEWAY_ACCOUNTS || '').split(',')
 
-const appendChargeForNewView = async function appendChargeForNewView (charge, req, chargeId) {
+/*
+  we want to enable wallet payments in all circumstances for our test payment gateways
+  if the gateway account id is any of our test payment gateways then return true
+  for any other account respect the values of the wallet toggles
+ */
+const walletEnabled = (walletToggle, gatewayAccountId, payTestGatewayAccounts) => {
+  try {
+    if (payTestGatewayAccounts.includes(gatewayAccountId.toString()) && !walletToggle) {
```

Comment:
Do we need this `&& !walletToggle` bit? Can't we always return true if the gateway account id is in the `PAY_TEST_GATEWAY_ACCOUNTS` list?

Reply:
because there's a logline for when this happens (as it should be unusual). Without it, the log would be present for every charge processed for the test gateways even if the feature toggles are switched on.